pkg/metrics/metrics.go

```diff
@@ -212,3 +208,16 @@ func EmitCSVMetric(oldCSV *olmv1alpha1.ClusterServiceVersion, newCSV *olmv1alpha
 		csvAbnormal.WithLabelValues(newCSV.Namespace, newCSV.Name, newCSV.Spec.Version.String(), string(newCSV.Status.Phase), string(newCSV.Status.Reason)).Set(1)
 	}
 }
+
+func EmitSubsMetrics(sub *olmv1alpha1.Subscription) {
+	// sub.Spec is not a required field.
+	if sub.Spec == nil {
+		return
+	}
+
+	SubscriptionSyncCount.WithLabelValues(sub.GetName(), sub.Status.InstalledCSV, sub.Spec.Channel, sub.Spec.Package).Inc()
+}
+
+func DeleteSubsMetric(sub *olmv1alpha1.Subscription) {
+	SubscriptionSyncCount.DeleteLabelValues(sub.GetName(), sub.Status.InstalledCSV, sub.Spec.Channel, sub.Spec.Package)
+}
```

Comment:
Can we add a test to the [metrics-e2e test](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/test/e2e/metrics_e2e_test.go)? The test could:
- Create a sub
- check for the metric
- update the sub
- check for the metric
- delete the sub
- ensure the metrics does not exist

Reply:
@awgreene added an e2e test case here: https://github.com/operator-framework/operator-lifecycle-manager/pull/1519/files#diff-41336190eb5305cb6fc38321ea468723R70-R84